src/features/todo/todoSlice.jsx

```diff
@@ -1,7 +1,19 @@
 import { createSlice } from '@reduxjs/toolkit'
+import { useEffect, useState } from 'react';
+const products = JSON.parse(localStorage.getItem('products')) || [];
 
+
+const calculateTotalCost = (products) => {
+    return products.reduce((accumulator, product) => {
+        const costString = product.info.cost;
+        const costWithoutDots = costString.replace(/\./g, '');
+        const cost = parseInt(costWithoutDots) * parseInt(product.number);
+        return accumulator + cost;
+    }, 0);
+};
 const initialState = {
-    products: JSON.parse(localStorage.getItem('products')) || []
+    products: products,
+    totalCost: calculateTotalCost(products) 
```

Comment:
`totalCost` should not be saved in state. this should be derived using selectors